IBPSA/Fluid/HeatPumps/ModularReversible/BaseClasses/PartialReversibleRefrigerantMachine.mo

```diff
@@ -0,0 +1,659 @@
+within IBPSA.Fluid.HeatPumps.ModularReversible.BaseClasses;
+partial model PartialReversibleRefrigerantMachine
+  "Grey-box model for reversible heat pumps and chillers using modular models to simulate the refrigerant cycle"
+  extends IBPSA.Fluid.Interfaces.PartialFourPortInterface(
+    redeclare final package Medium1 = MediumCon,
+    redeclare final package Medium2 = MediumEva,
+    final m1_flow_nominal=mCon_flow_nominal,
+    final m2_flow_nominal=mEva_flow_nominal,
+    final allowFlowReversal1=allowFlowReversalCon,
+    final allowFlowReversal2=allowFlowReversalEva,
+    final m1_flow_small=1E-4*abs(mCon_flow_nominal),
+    final m2_flow_small=1E-4*abs(mEva_flow_nominal));
+
+  //General
+  replaceable package MediumCon =
+    Modelica.Media.Interfaces.PartialMedium "Medium at sink side"
+    annotation (Dialog(tab = "Condenser"),choicesAllMatching=true);
+  replaceable package MediumEva =
+    Modelica.Media.Interfaces.PartialMedium "Medium at source side"
+    annotation (Dialog(tab = "Evaporator"),choicesAllMatching=true);
+  replaceable PartialModularRefrigerantCycle refCyc constrainedby
+    PartialModularRefrigerantCycle(final use_rev=use_rev)
+    "Model of the refrigerant cycle" annotation (Placement(transformation(
+          extent={{-18,-18},{18,18}}, rotation=90)));
+  parameter Modelica.Units.SI.HeatFlowRate QUse_flow_nominal
+    "Nominal heat flow rate at useful side of the device"
+    annotation (Dialog(group="Nominal Design"));
+
+  parameter Real y_nominal "Nominal relative compressor speed"
+    annotation (Dialog(group="Nominal Design"));
+  parameter Modelica.Units.SI.HeatFlowRate PEle_nominal
+    "Nominal electrical power, defined
+    based on QUse_flow_nominal and nominal conditions"
+    annotation (Dialog(group="Nominal Design"));
```

Comment:
Isn't then the `final` keyword more fitting? The parameter may be used in external models, this would not be possible if it is protected, or?
The parameter is set final in each `ModularReversible.mo`, e.g. here:
```
    final PEle_nominal=refCyc.refCycHeaPumHea.PEle_nominal,
```

Reply:
This then falls into my second point. The description "defined based on..." is restrictive. If it can be modified in other ways by other classes, this extra bit of explanation is then not necessary. Just "Nominal electrical power" is enough.